components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java

```diff
@@ -602,7 +603,8 @@ public void shareApplication(String ownerOrgId, String sharedOrgId, ServiceProvi
             }
             String sharedApplicationId;
             try {
-                ServiceProvider delegatedApplication = prepareSharedApplication(mainApplication, createdOAuthApp);
+                ServiceProvider delegatedApplication =
+                        prepareSharedApplication(mainApplication, createdOAuthApp, ownerOrgId, sharedTenantDomain);
```

Comment:
```suggestion
                ServiceProvider delegatedApplication =
                        prepareSharedApplication(mainApplication, createdOAuthApp, sharedOrgId);
```
According to the logic, additionally we requires mainApp's tenant domain, which can be take from `mainApplication` itself, and the shared-org-id which need to be added after /o/<org-id>. Hence we can pass the sharedOrgId instead the `sharedTenantDomain` though they seems equal.

Reply:
Fixed